### PR TITLE
REC-25: Backend recipe management

### PR DESCRIPTION
**Pull request by bremade**

### CODE REVIEW

File: api/Recipe.go [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+    }
+
+    if err != nil {
+        return err
+    }
+
+    return nil
+}
+
+func (api *Api) CheckTags(tagsInput []model.Tag) error {
```
[the remark this answers is not in the document]
Comment (bremade) [reply api/Recipe.go]: done

File: api/Recipe.go [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+    recipe, err := api.db.GetSingleRecipe(id)
+
+    if err != nil {
+        return false, err
+    }
+
+    return util.ContainsString(recipe.Creators, username), nil
+}
+
+func (api *Api) CheckIngredients(ingredientsInput []model.Ingredient) error {
```
[the remark this answers is not in the document]
Comment (bremade) [reply api/Recipe.go]: done

File: api/Api.go [hunk 3]
```diff
@@ -21,3 +22,19 @@ func New(db *persistence.DB) *Api {
 func (api *Api) Status(c *gin.Context) {
     c.JSONP(http.StatusOK, gin.H{ "status": "ok" })
 }
+
+func (api *Api) CheckLogin(c *gin.Context) (bool, string) {
```
[the remark this answers is not in the document]
Comment (bremade) [reply api/Api.go]: done

**Comment by bremade** [comment]
> Greak work! I would make the 3 marked functions private.

Adapted the methods

**Comment by matthinc** [comment]
LGTM

**Pull request merged**